packages/wdio-sync/src/index.js

```diff
@@ -83,23 +84,23 @@ const executeAsync = function (fn, repeatTest = 0, args = []) {
     }
 
     /**
-     * if we don't retry just return result
+     * handle promise response
      */
-    if (repeatTest === 0 || !result || typeof result.catch !== 'function') {
-        return new Promise(resolve => resolve(result))
+    if (result && typeof result.catch === 'function') {
+        return result.catch((e) => {
+            if (repeatTest) {
+                return executeAsync(fn, --repeatTest, args)
+            }
+
+            e.stack = e.stack.split('\n').filter(STACKTRACE_FILTER_FN).join('\n')
+            return Promise.reject(e)
+        })
     }
 
     /**
-     * handle promise response
+     * if we don't retry just return result
      */
-    return result.catch((e) => {
-        if (repeatTest) {
-            return executeAsync(fn, --repeatTest, args)
-        }
-
-        e.stack = e.stack.split('\n').filter(STACKTRACE_FILTER_FN).join('\n')
-        return Promise.reject(e)
-    })
+    return new Promise(resolve => resolve(result))
```

Comment:
Is it possible to make the function `async` instead of using the Promise object? Then you can just `return result` and for the rejection throw an error.

Reply:
This actually simplifies it a lot, the only problem is it makes it hard to differentiate between the two error cases, one when `fn.apply(this, args)` throws an error, and one when it returns a promise which rejects. At the moment we only filter the stack trace in the second case. 

Is it ok to filter it in both cases? Or is there a way to tell the two cases apart in the catch block?

Reply:
Yes, sounds good.
I think executeSync works like this, it was changed to async function recently